packages/client/src/components/tag/view/TagGridView.component.tsx

```diff
@@ -72,11 +72,35 @@ export const TagGridView: React.FC<TagGridViewProps> = ({ tags, study }) => {
     })
     .flat();
 
+  const [removeTag] = useRemoveTagMutation();
+
+  const handleRedo = (row: any) => {
+    console.log(`Redoing tag for tag ID: ${row._id}`);
+    removeTag({ variables: { tag: row._id } });
```

Comment:
After the tag is removed, the list of tags should be refetched to reflect the changes. I'd suggest adding another property to this component that is a function to refetch the tags.